feature-libs/product-configurator/rulebased/components/attribute/types/checkbox-list/configurator-attribute-checkbox-list.component.ts

```diff
@@ -1,69 +1,159 @@
 import { ConfigFormUpdateEvent } from '../../../form/configurator-form.event';
 import { ConfiguratorStorefrontUtilsService } from '../../../service/configurator-storefront-utils.service';
-import { ConfiguratorAttributeBaseComponent } from '../base/configurator-attribute-base.component';
+import { ConfiguratorAttributeQuantityService } from '../../quantity/configurator-attribute-quantity.service';
+import { ConfiguratorAttributeMultiSelectionBaseComponent } from '../base/configurator-attribute-multi-selection-base.component';
+
 @Component({
   selector: 'cx-configurator-attribute-checkbox-list',
   templateUrl: './configurator-attribute-checkbox-list.component.html',
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class ConfiguratorAttributeCheckBoxListComponent
-  extends ConfiguratorAttributeBaseComponent
+  extends ConfiguratorAttributeMultiSelectionBaseComponent
   implements OnInit {
-  @Input() attribute: Configurator.Attribute;
+  attributeCheckBoxForms = new Array<FormControl>();
+
   @Input() group: string;
-  @Input() ownerKey: string;
 
-  @Output() selectionChange = new EventEmitter<ConfigFormUpdateEvent>();
+  // TODO(#11681): make quantityService a required dependency and remove deprecated constructor
+  /**
+   * default constructor
+   * @param {ConfiguratorStorefrontUtilsService} configUtilsService
+   * @param {ConfiguratorAttributeQuantityService} quantityService
+   */
+  constructor(
+    // eslint-disable-next-line @typescript-eslint/unified-signatures
+    configUtilsService: ConfiguratorStorefrontUtilsService,
+    // eslint-disable-next-line @typescript-eslint/unified-signatures
+    quantityService: ConfiguratorAttributeQuantityService
+  );
+
+  /**
+   * @deprecated since 3.3
+   */
+  constructor(configUtilsService: ConfiguratorStorefrontUtilsService);
 
   constructor(
-    protected configUtilsService: ConfiguratorStorefrontUtilsService
+    protected configUtilsService: ConfiguratorStorefrontUtilsService,
+    protected quantityService?: ConfiguratorAttributeQuantityService
   ) {
-    super();
+    super(quantityService);
   }
 
-  attributeCheckBoxForms = new Array<FormControl>();
+  ngOnInit(): void {
+    const disabled = !this.allowZeroValueQuantity;
 
-  ngOnInit() {
     for (const value of this.attribute.values) {
       let attributeCheckBoxForm;
+
       if (value.selected === true) {
-        attributeCheckBoxForm = new FormControl(true);
+        attributeCheckBoxForm = new FormControl({
+          value: true,
+          disabled: disabled,
+        });
       } else {
         attributeCheckBoxForm = new FormControl(false);
       }
       this.attributeCheckBoxForms.push(attributeCheckBoxForm);
     }
   }
 
-  /**
-   * Triggered when a value is selected
-   */
+  get allowZeroValueQuantity(): boolean {
+    return (
+      this.quantityService?.allowZeroValueQuantity(this.attribute) ?? false
+    );
+  }
+
   onSelect(): void {
```

Comment:
This function and the `onSelect` in `ConfiguratorAttributeMultiSelectionBundleComponent` do the same thing. So, in here, we also can pass `value.valueCode` from html. And `assembleValuesForMultiSelectAttributes` can be removed or deprecated.

Reply:
We also can move this `onSelect(valueCode, state)` into parent class.

Reply:
@AlexanderKuschner, could you have a look at it?

Reply:
Picked

Reply:
As discussed: addressed with #12057